ERCS/erc-7920.md

```diff
@@ -0,0 +1,358 @@
+        bytes32 message = keccak256(
+            abi.encode(
+                "\x19\x01",
+                DOMAIN_SEPARATOR,
+                keccak256(abi.encode(MESSAGE_TYPEHASH, orderId, user))
+            )
+        );
+
+        if (
+            !_verifyCompositeSignature(
+                message,
+                proof,
+                merkleRoot,
+                signature,
+                user
+            )
+        ) {
+            revert Unauthorized();
+        }
+
+        // DO STUFF
+    }
+
+    function _verifyCompositeSignature(
+        bytes32 message,
+        bytes32[] calldata proof,
+        bytes32 merkleRoot,
+        bytes calldata signature,
+        address expectedSigner
+    ) internal view returns (bool) {
+        if (!_verifyMerkleProof(message, proof, merkleRoot)) {
+            return false;
+        }
+
+        return recover(merkleRoot, signature) == expectedSigner;
+    }
+
+    function _verifyMerkleProof(
+        bytes32 leaf,
+        bytes32[] calldata proof,
+        bytes32 root
+    ) internal pure returns (bool) {
+        bytes32 computedRoot = leaf;
+        for (uint256 i = 0; i < proof.length; ++i) {
+            if (computedRoot < proof[i]) {
+                computedRoot = keccak256(abi.encode(computedRoot, proof[i]));
+            } else {
+                computedRoot = keccak256(abi.encode(proof[i], computedRoot));
+            }
+        }
+        return computedRoot == root;
+    }
+
+    function recover(
+        bytes32 digest,
+        bytes memory signature
+    ) internal pure returns (address) {
+        require(signature.length == 65, "Invalid signature length");
+
+        bytes32 r;
+        bytes32 s;
+        uint8 v;
+
+        assembly {
+            r := mload(add(signature, 32))
+            s := mload(add(signature, 64))
+            v := byte(0, mload(add(signature, 96)))
+        }
+
+        return ecrecover(digest, v, r, s);
+    }
+}
+```
+
+### Specification of `eth_signTypedData_v5` JSON RPC method.
+
+This ERC adds a new method `eth_signTypedData_v5` to Ethereum JSON-RPC. This method allows signing multiple typed data messages with a single signature using the specification described above. The signing account must be prior unlocked.
+
+This method returns: the signature, merkle root, and an array of proofs (each corresponding to an input message).
+
+#### Parameters
+
+1. `Address` - Signing account
+2. `TypedData | TypedDataArray` - A single TypedData object or Array of `TypedData` objects from EIP-712.
+
+##### Returns
+
+```JavaScript
+{
+  signature: 'DATA',     // Hex encoded 65 byte signature (same format as eth_sign)
+  merkleRoot: 'DATA',    // 32 byte Merkle root as hex string
+  proofs: [              // Array of Merkle proofs (one for each input message)
+    ['DATA', 'DATA'], // First message proof (array of 32 byte hex strings)
+    ['DATA', 'DATA'], // Second message proof
+    ...
+  ]
+}
+```
+
+##### Example
+
+Request:
+
+```shell
```

Comment:
Instead of wrapping in curl, would it maybe be easier to read if you only provide the JSON input? Running it through a formatter (eg. `jq .`) might help too.